crypto/fipsmodule/rand/rand.c

```diff
@@ -382,6 +387,9 @@ void RAND_bytes_with_additional_data(uint8_t *out, size_t out_len,
   const uint64_t fork_generation = CRYPTO_get_fork_generation();
   const int fork_unsafe_buffering = rand_fork_unsafe_buffering_enabled();
 
+  uint32_t snapsafe_generation = 0;
+  int snapsafe_status = CRYPTO_get_snapsafe_generation(&snapsafe_generation);
```

Comment:
what if this is 0 after the call to `CRYPTO_get_snapsafe_generation`? is that a case we need to handle as on L558 or retry?

Reply:
It's expected that `snapsafe_generation` will likely be 0 after the call to `CRYPTO_get_snapsafe_generation` -- in fact, that'll be true for any environment that doesn't have the `/dev/sysgenid` setup in the kernel.  What we care about is whether the `snapsafe_generation` has *changed* since the last time we re-seeded the DRBG.